test/common/router/config_impl_test.cc

```diff
@@ -5081,8 +5081,7 @@ TEST_F(BadHttpRouteConfigurationsTest, BadRouteEntryConfigPrefixAndPath) {
 
   EXPECT_THROW_WITH_REGEX(
       TestConfigImpl(parseRouteConfigurationFromYaml(yaml), factory_context_, true), EnvoyException,
-      "invalid value oneof field 'path_specifier' is already set. Cannot set '(prefix|path)' for "
-      "type oneof");
+      "invalid value oneof field 'path_specifier' is already set.");
```

Comment:
So part of the message is missing when running on windows?

Reply:
Yes. For this example specifically, we get "invalid value oneof field 'path_specifier' is already set. Cannot set 'prefix' for type oneof"

Reply:
Isn't the `(prefix|path)` part a regex capture group for the `EXPECT_THROW_WITH_REGEX`? I looks like this failure is a symptom of googletest not supporting more advanced regex

Running this test locally I see:

```
[ RUN      ] BadHttpRouteConfigurationsTest.BadRouteEntryConfigPrefixAndPath
external/com_google_googletest/googletest/src/gtest-port.cc(860): error: Failed
Syntax error at index 71 in simple regular expression "invalid value oneof field 'path_specifier' is already set. Cannot set '(prefix|path)' for type oneof":
'(' is unsupported.
Stack trace:
...
```

Reply:
@sunjayBhatia is right. I've updated the PR to use GTEST_USES_SIMPLE_RE as we have done in other parts of the codebase, so that this can be simplified if/when google test supports capture groups Windows.

Reply:
Realistically the only solve is the adoption of RE2, why Google test doesn't support Google RE2 is a little puzzling :-)

We can in fact shift as a project to pcre if we don't believe this is going to happen any time soon.